### PR TITLE
Typings for react-bootstrap-typeahead (Fix: https://github.com/ericgio/react-bootstrap-typeahead/pull/128)

### PR DESCRIPTION
**Pull request by Guymestef**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-bootstrap-typeahead/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "es2015",
+        "lib": [
+            "es6",
+            "dom",
```
Comment (andy-ms) [inline types/react-bootstrap-typeahead/tsconfig.json]: trailing comma not allowed in JSON

Comment (Guymestef) [reply types/react-bootstrap-typeahead/tsconfig.json]: It's fixed

File: types/react-bootstrap-typeahead/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts",
+        "react-bootstrap-typeahead-tests.ts"
```
Comment (andy-ms) [inline types/react-bootstrap-typeahead/tsconfig.json]: `.tsx`

File: types/react-bootstrap-typeahead/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    renderMenuItemChildren?: (option: T, props: TypeaheadProps<T>, index: number) => any;
+    renderToken?: (selectedItem: T | string, onRemove: () => void) => any;
+    selected?: T[];
+    selectHintOnEnter?: boolean;
+    submitFormOnEnter?: boolean;
+}
+
+export const Typeahead: React.ClassicComponentClass<TypeaheadProps<any>>;
+
+export default Typeahead;
```
Comment (andy-ms) [inline types/react-bootstrap-typeahead/index.d.ts]: This doesn't seem to really be a default export. The [readme](https://github.com/ericgio/react-bootstrap-typeahead) recommends using `import {Typeahead} from 'react-bootstrap-typeahead';`, which is a named import (`import Typeahead from 'react-bootstrap-typeahead';` would be a default import.)

File: types/react-bootstrap-typeahead/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+// Type definitions for react-bootstrap-typeahead 2.0
+// Project: https://github.com/ericgio/react-bootstrap-typeahead
+// Definitions by: My Self <https://github.com/Guymestef>
```
Comment (andy-ms) [inline types/react-bootstrap-typeahead/index.d.ts]: Needs to fill in `My Self`, then should be good to merge

Comment (Guymestef) [reply types/react-bootstrap-typeahead/index.d.ts]: Well I don't really mind to be flagged as the one who provided the typings but My Self replaced 😄

**Comment by andy-ms** [comment]
Thanks!

**Comment by Guymestef** [comment]
Hi @andy-ms 
There's still no typings @types/react-bootstrap-typeahead on npm.
Is it normal and we just need to wait longer?

**Pull request merged**